### DISCUSSION TITLE
Telemetry data when power off

### DISCUSSION THREAD

**Discussion by isorin** [arendst/Tasmota#11709] [Q&A] [answered]
Is telemetry data pushed even when SetOption21(Energy monitoring when power is off) is 0 (disabled) and power is off?

The console shows:

```
13:49:03.563 MQT: tele/tasmota_B5B44B/SENSOR = {"Time":"2021-04-13T13:49:03","ENERGY":{"TotalStartTime":"2020-06-12T00:41:10","Total":144.00708,"Yesterday":0.90450,"Today":0.00141,"Period":0.0,"Power":0.0,"ApparentPower":0.0,"ReactivePower":0.0,"Factor":0.00,"Voltage":0,"Current":0.000}}
13:51:03.547 MQT: tele/tasmota_B5B44B/SENSOR = {"Time":"2021-04-13T13:51:03","ENERGY":{"TotalStartTime":"2020-06-12T00:41:10","Total":144.00708,"Yesterday":0.90450,"Today":0.00141,"Period":0.0,"Power":0.0,"ApparentPower":0.0,"ReactivePower":0.0,"Factor":0.00,"Voltage":0,"Current":0.000}}
13:53:03.530 MQT: tele/tasmota_B5B44B/SENSOR = {"Time":"2021-04-13T13:53:03","ENERGY":{"TotalStartTime":"2020-06-12T00:41:10","Total":144.00708,"Yesterday":0.90450,"Today":0.00141,"Period":0.0,"Power":0.0,"ApparentPower":0.0,"ReactivePower":0.0,"Factor":0.00,"Voltage":0,"Current":0.000}}

```
Is this expected?

**Comment by ascillato** [comment, marked as answer]
yes, as designed

**Reply by sfromis** [reply]
Indeed, SetOption21 only controls if it keeps the energy monitoring chip active while output power is off, providing voltage and frequency. As these are 0, you have successfully disabled the monitoring :-)

**Reply by isorin** [reply]
Perfect, thank you!

**Reply by ascillato** [reply]
Exactly. Teleperiod is independent of any option and it is designed to report every *teleperiod* seconds.

**Reply by ascillato** [reply]
If you want, you can disable teleperiod, but the point of teleperiod is to report independently of what is happening. It is used by several automation softwares. So the use of teleperiod, depends on your own use-case.

**Reply by sfromis** [reply]
If you want reporting of energy monitoring data when things change, [PowerDelta](https://tasmota.github.io/docs/Commands/#powerdelta) is a good way of doing it. May be more useful than the cyclic TelePeriod payloads.
